### PR TITLE
chore(js): specify exact versions for production dependencies

### LINKED ISSUE #5067: RFC: Specify exact JS dependencies in `package.json`s
# Overview

Our `dependencies` in our JS `package.json`s are (almost all) specified with `^` semver version ranges (e.g. `"redux": "^4.0.1"`). I think we should **specify them as exact versions instead**

# Reasoning

- Most of our projects are applications, not libraries
    - Semver versions are great for libraries where ranges can mean the end application has to install fewer dependencies because more stuff can satisfy ranges
- Our projects that are libraries are all internal, which means the advantage for semver ranges in libraries above doesn't really come into play
- Exact versions are easier to explicitly keep in sync across projects
- Our versions are already implicitly locked by our lockfile, so why pretend like we're not locking them?

## development dependencies

I don't have as strong an opinion about development dependencies, but several of the bullets above also apply. If we wanted to, we could lock development dependencies, too

### PR DESCRIPTION
**Pull request by mcous**
## overview

This PR changes all production dependency specs in `package.json` files from semver compatitble range (`^x.y.z`) to exact (`x.y.z`). Closes #5067, see ticket for reasoning.

## changelog

- chore(js): specify exact versions for production dependencies

## review requests

Examine the lockfile closely. The goal of this PR is to **avoid changing the actual dependency used** and simply make the package.json more accurately reflect reality.

## risk assessment

Low assuming now actual dependency usage has changed

### CODE REVIEW

File: labware-library/package.json [hunk 1]
```diff
@@ -17,20 +17,20 @@
   "dependencies": {
     "@hot-loader/react-dom": "16.8.6",
     "@opentrons/components": "3.18.0",
-    "ajv": "^6.2.1",
```
Comment (mcous) [inline labware-library/package.json:20]: According to the lockfile, this was already resolving to 6.10.2, hence the change to align it with all our other projects that use ajv

File: protocol-designer/package.json [hunk 2]
```diff
@@ -22,31 +22,28 @@
...
+    "react-redux": "7.1.0",
     "redux": "4.0.5",
-    "redux-actions": "^2.2.1",
-    "redux-thunk": "^2.2.0",
-    "reselect": "^4.0.0",
-    "uuid": "^3.3.2",
-    "yup": "^0.26.6"
-  },
-  "devDependencies": {
-    "reselect-tools": "^0.0.7"
```
Comment (mcous) [inline protocol-designer/package.json:50]: Moved to repo-level development dependencies

File: yarn.lock [hunk 3]
```diff
@@ -3465,16 +3476,6 @@ ajv@^6.10.0, ajv@^6.9.1:
     json-schema-traverse "^0.4.1"
     uri-js "^4.2.2"
 
-ajv@^6.10.2, ajv@^6.2.1:
```
Comment (mcous) [inline yarn.lock:3468]: See comment above about `ajv@6.2.1`

**Review by IanLondon — APPROVED** [review]
thanks!

**Pull request merged**